transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/TransformationPipelineBuilder.scala

```diff
@@ -180,6 +179,82 @@ object TransformationPipelineBuilder extends PipelineBuilder[Args] {
         .groupByKey
     }
 
+    // Get analysis step objects
+    val analysisStepRuns = readRawEntities(EncodeEntity.AnalysisStepRun)
+    val analysisStepVersionsById = readRawEntities(EncodeEntity.AnalysisStepVersion)
+      .withName("Key analysis step versions by ID")
+      .keyBy(_.read[String]("@id"))
+    val analysisStepsById = readRawEntities(EncodeEntity.AnalysisStep)
+      .withName("Key analysis steps by ID")
+      .keyBy(_.read[String]("@id"))
+    val filesByStepRun = fileInputs
+      .withName("Key files by step run ID")
+      .keyBy(_.tryRead[String]("step_run").getOrElse(""))
+      .groupByKey
+
+    // Join AnalysisStepRuns, AnalysisStepVersions, AnalysisSteps, and Files.
+    // They will be used in both the StepRun and PipelineRun transformations.
+    val stepRunInfo = analysisStepRuns
+      .withName("Key step runs by analysis step version")
+      .keyBy(_.read[String]("analysis_step_version"))
+      .join(analysisStepVersionsById)
+      .values
+      .withName("Key step runs by analysis step")
+      .keyBy(_._2.read[String]("analysis_step"))
+      .join(analysisStepsById)
+      .values // outputs ((stepRun, stepVersion), step)
+      .withName("Key step runs by step run ID")
+      .keyBy(_._1._1.read[String]("@id"))
+      .leftOuterJoin(filesByStepRun)
+      .values
+      .withSideInputs(fileIdToType)
+      .withName("Transform step runs")
+
+    // Transform step runs
+    val stepRunOutput = stepRunInfo.map {
+      case ((((stepRun, stepVersion), step), generatedFiles), sideCtx) =>
+        StepRunTransformations.transformStepRun(
+          stepRun,
+          stepVersion,
+          step,
+          generatedFiles.toIterable.flatten,
+          sideCtx(fileIdToType)
+        )
+    }.toSCollection
+    StorageIO.writeJsonLists(
+      stepRunOutput,
+      "Step Runs",
+      s"${args.outputPrefix}/step_run"
+    )
+
+    // Transform pipeline runs
+    val pipelinesById = readRawEntities(EncodeEntity.Pipeline)
+      .withName("Key pipelines by ID")
+      .keyBy(_.read[String]("@id"))
+
+    val pipelineRunOut = stepRunInfo.flatMap {
+      case ((((stepRun, _), step), generatedFiles), _) =>
+        val stepRunId = CommonTransformations.readId(stepRun)
+        val pipelineId = PipelineRunTransformations.getPipelineId(step)
+        val experimentId =
+          PipelineRunTransformations.getExperimentId(generatedFiles.toIterable.flatten, stepRunId)
+        if (pipelineId.isEmpty || experimentId.isEmpty) None
+        else Some((pipelineId.head, experimentId.head))
```

Comment:
This is duplicated from `StepRunTransformations`, yes? Could we extract it into a helper method?